src/lib/auth.ts

```diff
@@ -5,17 +5,37 @@ import type { User } from "firebase/auth"
 import { app } from './firebase'
 
 export interface AuthState {
+  // auth.user will be either a firebase User object or null
   user: User | null
+  // auth.known is whether or not we have completed trying to login with firebase
+  // --i.e. if `true` then we "know" that we have a user or we "know" that we don't,
+  // but when the store is first initialized we don't "know" anything.
```

Comment:
This is slightly misleading, it's not whether we have "completed trying to login with firebase", as we may not have tried at all, it's whether the auth state has settled or been resolved to know whether the user is signed in or not.